Makefile.am

```diff
@@ -74,6 +74,31 @@ V_TAR = $(V_TAR_$(V))
 V_TAR_ = $(V_TAR_$(AM_DEFAULT_VERBOSITY))
 V_TAR_0 = @echo "  TAR     " $@;
 
+# We set SKIP_NODE to 1 if we should avoid buidling anything that
+# depends on node_modules/ having been created.  This allows doing
+# variable expansions like x_DATA = $(FILES$(SKIP_NODE)) which will
+# expand to $(FILES) in the normal case, and to the empty variable
+# $(FILES1) in the case SKIP_NODE is set.  This is similar to the tricks
+# used by automake for verbosity control (ie: V=1 et al).
+#
+# We give the user explicit control over this process via the NO_NPM
+# make variable.  As a user you should use `make NO_NPM=1` or `make
+# NO_NPM=0` to force the value one way or the other.  Otherwise, we
+# guess, depending on the presence of the `node_modules` directory.
+SKIP_NODE := $(shell "$(top_srcdir)/tools/should-skip-node" "$(NO_NPM)")
```

Comment:
Any chance you could pick *one* variable name (SKIP_NODE vs. NO_NPM)?

Reply:
One of these is the user variable (`NO_NPM`) and one of them is an internal-only variable (`SKIP_NODE`).  `SKIP_NODE` can be set according to the value of `NO_NPM` *or* detection of the `node_modules` dir.  I could give the internal variable a "more internal" name to make that more obvious, I suppose...